src/nimodinst/templates/session.py.mako

```diff
@@ -64,8 +64,8 @@ class Device(object):
             return object.__getattribute__(self, name).__getitem__(None)
 
     def __setattr__(self, name, value):
-        if self._is_frozen and name not in ['_is_frozen', 'index']:
-            raise TypeError("%s is not writable" % name)
+        if self._is_frozen:
+            raise AttributeError("Set attribute not supported.")
```

Comment:
This is misleading.

* Text is hard to decifer "Set attribute not supported."
* You get the same error for a real attribute as you do for "potato"

Reply:
Should I change the text? "Set attribute is not a supported function."

Also per our discussion yesterday, if a customer tries to set any attribute (real or other wise) they will get the above error. I feel that this makes sense and gives the customer an error that they would expect. While I agree it might be nice to say "potato is not an attribute, and you also can't set attributes", the difficulty to do this is high due to the dir() issues discussed earlier, and the value add is very minimal.

 If a customer tries to call "Get Attribute" with potato they will get an expected error that says that the attribute does not exist (which is tested in test_modinst.py).

Reply:
Fixed per our discussion